### PR TITLE
cares: enable z/OS support

### PR DESCRIPTION
**Pull request by gabylb**
- remove -g
- adjust cflags for z/OS
- add z/OS configuration
- include sys/select.h for z/OS

As well as the following (consider separate PR(s) for master.zos):
- add z/OS implementation of retrieving DNS server list
- change default_malloc(0) to malloc(1) for z/OS
- add header for strncasecmp to use proper version

##### Checklist
- [x] `make cares` passes
- [x] commit message follows [commit guidelines]

### CODE REVIEW

File: deps/cares/cares.gyp [hunk 1]
```diff
@@ -136,15 +136,14 @@
         }, {
           # Not Windows i.e. POSIX
           'cflags': [
-            '-g',
```
Comment (zsw007) [inline deps/cares/cares.gyp:139]: This looks like a change that applies to other platforms as well. Was the -g flag left in by mistake by the Node.js team at some point? If so, I think we might want to split this into a separate PR since it's not strictly a z/OS change. Alternatively, we can leave it in for now and split it when we push to `master.zos`.

**Review by zsw007 — APPROVED** [review]
Do all of these fit into a single PR? For example, would it make sense to have 1 PR to enable cares to build with the correct configurations, and then another PR to add in the missing z/OS implementation of retrieving DNS server list? 

If they all go together, then maybe we can adjust the PR title to say something more descriptive like `enable z/OS support`.

Alternatively, we can leave it in for now and make these changes for when we push to `master.zos`.

What do you guys think?

LGTM otherwise.

**Comment by IgorTodorovskiIBM** [comment]
I agree, changing it to "cares: enable z/OS support" is more descriptive

File: deps/cares/cares.gyp (hunk 1 above)
Comment (IgorTodorovskiIBM) [reply deps/cares/cares.gyp:139]: Or we could do `'cflags!' : ['-g']` in the z/OS guard below so as not to affect other platforms

Comment (gabylb) [reply deps/cares/cares.gyp:139]: That's what I thought (that it must've been left by mistake), so I left the 'remove -g' separated as in 'remove -g, and adjust cflags for z/OS'

I think let's leave it for now, and separate it into another PR for master.zos.

Comment (gabylb) [reply deps/cares/cares.gyp:139]: Separated the 'remove -g' in the description for clarity.

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM

**Review by alexcfyung — APPROVED** [review]
LGTM
